File: test/benchmark/ruby/trees.rb
class Tree
    attr_accessor :depth, :a, :b, :c, :d, :e
    def initialize(depth)
        @depth = depth
        if depth > 0
            @a = Tree.new(depth - 1)
            @b = Tree.new(depth - 1)
            @c = Tree.new(depth - 1)
            @d = Tree.new(depth - 1)
            @e = Tree.new(depth - 1)
        end
    end

    def walk()
        if @depth == 0
            return 0
        end

        return @depth + @a.walk() + @b.walk() + @c.walk() + @d.walk() + @e.walk()
    end
end

tree = Tree.new(8)
start = Time.now
for i in (1..100)
    if (tree.walk() != 122068)
        puts("Error")
    end
end

puts(Time.now - start)

File: test/benchmark/python/trees.py
import time


class Tree:
    def __init__(self, depth):
        self.depth = depth
        if depth > 0:
            self.a = Tree(depth - 1)
            self.b = Tree(depth - 1)
            self.c = Tree(depth - 1)
            self.d = Tree(depth - 1)
            self.e = Tree(depth - 1)

    def walk(self):
        if self.depth == 0:
            return 0

        return (
            self.depth
            + self.a.walk()
            + self.b.walk()
            + self.c.walk()
            + self.d.walk()
            + self.e.walk()
        )


tree = Tree(8)
start = time.process_time()
for i in range(100):
    if tree.walk() != 122068:
        print("Error")

print(time.process_time() - start)
